rlog/2024-06-24-gsub-largemsg-improvements.mdx

```diff
@@ -0,0 +1,430 @@
+it learns about the new message and can request it by sending an IWANT request without waiting for the heartbeat (gossip) interval. 
+As a result, a significantly smaller number of transmissions is sufficient for propagating the message to the entire network. 
+
+This approach conserves peer bandwidth by minimizing redundant transmissions 
+while ensuring Gossipsub resilience at the cost of one RTT (for missing peers). 
+Interestingly, Curtailing queuing delays can also help lower network-wide message dissemination latency (for huge messages).
+
+However, finding an appropriate value for $K$ is crucial for optimal performance. 
+A smaller $K$ saves peer bandwidth, while a larger $K$ achieves quicker spread until outgoing message queues pile up. 
+Setting $K = D_{low}$ can be one option. 
+
+It is worth mentioning that such behavior may negatively impact peer scoring (by missing message delivery rewards from $D-K$ peers). 
+However, a minimized workload enables early message dissemination to the remaining peers. 
+These early transmissions and randomized $K$ set selection can help achieve an overall better peer score.
+
+### 4. Message Prioritization
+Despite the standardized specifications of the GossipSub protocol, 
+the message forwarding mechanisms can significantly impact network-wide message dissemination latency and bandwidth utilization. 
+
+It is worth mentioning that every node is responsible for transmitting different types of packets, 
+including control messages, locally published messages, messages received from mesh members, IWANT replies, etc. 
+
+As long as traffic volume is lower than the available data rate, 
+the message forwarding mechanisms yield similar results due to negligible queuing delays. 
+
+However, when the traffic volume increases and exceeds the available peer bandwidth (even for short traffic bursts), 
+the outgoing message queue(s) sizes rise, potentially impacting the network's performance. 
+
+In this scenario, FIFO-based traffic forwarding can lead to locally published messages being placed at the end of the outgoing message queue, 
+introducing a queuing delay proportional to the queue size. 
+The same applies to other delay-sensitive messages like IDONTWANT, PRUNE, etc. 
+
+On the other hand, the segregation of traffic into priority and non-priority queues can potentially starve low-priority messages. 
+One possible solution is to use weighted queues for a fair spread of messages.
+
+Message prioritization can be a powerful tool to ensure that important messages reach their intended recipients on time 
+and allow for customizable message handling. 
+
+For example, staggering between peers and messages can be better managed by using priority queues. 
+However, it's important to note that message prioritization also introduces additional complexity to the system, 
+necessitating sophisticated algorithms for better message handling.   
+
+### 5. Better Benefits from IWANT/FloodPublish

```

Comment:
```suggestion
### 5. Better benefits from IWANT/FloodPublish

```

Reply:
"Better benefits" reads quite clumsily to me. I'd consider "Additional benefits from...." or similar wording - not a major thing either way though